### PR TITLE
[#2695] feat(doc): Add docs for fileset catalog

### PR DESCRIPTION
**Pull request by jerryshao**
### What changes were proposed in this pull request?

This PR proposes to add docs for fileset catalog.

### Why are the changes needed?

Fix: #2695 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

No.

### CODE REVIEW

**Comment by jerryshao** [comment]
@coolderli @xloya would you please help to review? Thanks.

**Comment by coolderli** [comment]
@jerryshao Do we need to introduce how to use the Fileset in the Spark engine? In addition, I have already tested the Tensorflow and submitted an MR: https://github.com/tensorflow/io/pull/1970. After https://github.com/datastrato/gravitino/pull/2779 is resolved, we can support tensorflow. I think we can add a doc like https://help.aliyun.com/zh/hdfs/using-tensorflow-on?spm=a2c4g.11186623.0.i6. What do you think?

**Comment by jerryshao** [comment]
> @jerryshao Do we need to introduce how to use the Fileset in the Spark engine? In addition, I have already tested the Tensorflow and submitted an MR: [tensorflow/io#1970](https://github.com/tensorflow/io/pull/1970). After #2779 is resolved, we can support tensorflow. I think we can add a doc like https://help.aliyun.com/zh/hdfs/using-tensorflow-on?spm=a2c4g.11186623.0.i6. What do you think?

I would suggest to have another doc about gvfs and add Spark, TF related things there.

File: docs/hadoop-catalog.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+keyword: hadoop catalog
+license: "Copyright 2024 Datastrato Pvt Ltd.
+This software is licensed under the Apache License version 2."
+---
+
+## Introduction
+
+Hadoop catalog is a fileset catalog that using Hadoop Compatible File System (HCFS) to manage
+the storage location of the fileset. Currently, it supports local filesystem and HDFS. For
+object stores like S3, ADLS, and GCS, we haven't yet tested.
```
Comment (qqqttt123) [inline docs/hadoop-catalog.md:14]: Should we avoid using `we`?

File: docs/hadoop-catalog.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+### Schema operations
+
+Refer to [Schema operation](./manage-fileset-metadata-using-gravitino.md#schema-operations) for more details.
+
+## Fileset
+
+### Fileset capabilities
+
+- The Hadoop catalog supports creating, updating, and deleting filesets.
```
Comment (qqqttt123) [inline docs/hadoop-catalog.md:52]: list filesets?

File: docs/hadoop-catalog.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+### Fileset properties
+
+No.
```
[the remark this answers is not in the document]
Comment (jerryshao) [reply docs/hadoop-catalog.md:56]: All the fileset operations are already defined in the manage-fileset-metadata.md, And there's no difference for Hadoop catalog, so I was thinking that it is not necessary to write here again.

[the remark this answers is not in the document]
Comment (mchades) [reply docs/hadoop-catalog.md:56]: If so, should we put a link here?

File: docs/hadoop-catalog.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+## Fileset
+
+### Fileset capabilities
+
+- The Hadoop catalog supports creating, updating, deleting, and listing filesets.
+
+### Fileset properties
+
+No.
```
Comment (coolderli) [inline docs/hadoop-catalog.md:59]: Change `No` to `None`?

File: docs/hadoop-catalog.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+Hadoop catalog is a fileset catalog that using Hadoop Compatible File System (HCFS) to manage
+the storage location of the fileset. Currently, it supports local filesystem and HDFS. For
+object storage like S3, GCS, and Azure Blob Storage, you can put the hadoop object store jar like
+hadoop-aws into the `$GRAVITINO_HOME/catalogs/hadoop/libs` directory to enable the support.
+Gravitino itself haven't yest tested the object storage support, so if you have any issue,
+please create an [issue](https://github.com/datastrato/gravitino/issues).
+
+Note that the Hadoop catalog is built against Hadoop 3, it should be compatible with both Hadoop
+2.x and 3.x, since we don't leverage any new features in Hadoop 3. If there's any compatibility
```
Comment (qqqttt123) [inline docs/hadoop-catalog.md:20]: `we don't leverage` -> `Gravitino doesn't leverage`.

File: docs/manage-fileset-metadata-using-gravitino.md [hunk 6]
```diff
@@ -0,0 +1,422 @@
...
+tabular data and others in Gravitino with a unified way.
+
+After fileset is created, users can easily access, manage the files/directories through
+Fileset's identifier, without needing to know the physical path of the managed datasets. Also, with
+unified access control mechanism, filesets can also be managed via the same role based access
+control mechanism without needing to set access controls to different storages.
+
+To use fileset, we assume that:
+
+ - Gravitino server is launched, and the host and port is [http://localhost:8090](http://localhost:8090).
```
Comment (qqqttt123) [inline docs/manage-fileset-metadata-using-gravitino.md:27]: You launched a Gravitino server.

File: docs/hadoop-catalog.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+## Introduction
+
+Hadoop catalog is a fileset catalog that using Hadoop Compatible File System (HCFS) to manage
+the storage location of the fileset. Currently, it supports local filesystem and HDFS. For
+object storage like S3, GCS, and Azure Blob Storage, you can put the hadoop object store jar like
+hadoop-aws into the `$GRAVITINO_HOME/catalogs/hadoop/libs` directory to enable the support.
+Gravitino itself haven't yest tested the object storage support, so if you have any issue,
+please create an [issue](https://github.com/datastrato/gravitino/issues).
+
+Note that the Hadoop catalog is built against Hadoop 3, it should be compatible with both Hadoop
```
Comment (qqqttt123) [inline docs/hadoop-catalog.md:19]: Should we avoid passive sentence?

File: docs/manage-fileset-metadata-using-gravitino.md [hunk 8]
```diff
@@ -0,0 +1,422 @@
...
+
+:::tip
+ - Users should create a metalake, a catalog and a schema before creating a fileset.
+ - Current Gravitino only supports managing Hadoop Compatible File System (HCFS) locations.
+:::
+
+### Create a fileset
+
+You can create a fileset by sending a `POST` request to the `/api/metalakes/{metalake_name}
+/catalogs/{catalog_name}/schemas/{schema_name}/files` endpoint or just use the Gravitino Java
```
Comment (coolderli) [inline docs/manage-fileset-metadata-using-gravitino.md:227]: `/api/metalakes/{metalake_name}
/catalogs/{catalog_name}/schemas/{schema_name}/files`
--
Should it be `metalakes/{metalake}/catalogs/{catalog}/schemas/{schema}/filesets`? https://github.com/datastrato/gravitino/blob/15443039113ee3b8a14591feb262cccf6431f952/server/src/main/java/com/datastrato/gravitino/server/web/rest/FilesetOperations.java#L40

File: docs/manage-fileset-metadata-using-gravitino.md [hunk 9]
```diff
@@ -0,0 +1,422 @@
...
+
+</TabItem>
+</Tabs>
+
+Currently, Gravitino supports two **types** of the fileset:
+
+ - `MANAGED`: The storage location of the fileset is managed by Gravitino, when specified as
+   `MANAGED`, the physical location of the fileset will be deleted when this fileset is dropped.
+ - `EXTERNAL`: The storage location of the fileset is managed by users, when specified as
+   `EXTERNAL`, the physical location of the fileset will not be deleted when this fileset is
```
Comment (coolderli) [inline docs/manage-fileset-metadata-using-gravitino.md:282]: we can add a bold to **not**.

File: docs/manage-fileset-metadata-using-gravitino.md [hunk 10]
```diff
@@ -0,0 +1,422 @@
...
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+// ...
+Catalog catalog = gravitinoMetaLake.loadCatalog(NameIdentifier.of("metalake", "catalog"));
+
+FilesetCatalog filesetCatalog = catalog.asFilesetCatalog();
+NameIdentifier[] identifiers =
+    filesetCatalog.listFilesets(Namespace.ofFileset("metalake", "catalog", "schema"));
```
Comment (coolderli) [inline docs/manage-fileset-metadata-using-gravitino.md:417]: I think there's another issue. The `metalake` in Namespace seems redundant. The new GravitinoClient, we have declared the name of the current metalake. It is not related to this MR. Never mind.

**Comment by coolderli** [comment]
@jerryshao Left some comments. Overall, it looks good to me.

File: docs/manage-fileset-metadata-using-gravitino.md (hunk 10 above)
Comment (jerryshao) [reply docs/manage-fileset-metadata-using-gravitino.md:408]: Yeah, it is unrelated, and we are working on the client refactoring things.

**Comment by jerryshao** [comment]
@shaofengshi would you please also check the java client part? Thanks.

File: docs/manage-fileset-metadata-using-gravitino.md [hunk 11]
```diff
@@ -0,0 +1,413 @@
...
+Typically, a fileset is mapping to a directory on a file system like HDFS, S3, ADLS, GCS, etc.
+With fileset managed by Gravitino, the non-tabular data can be managed as assets together with
+tabular data and others in Gravitino with a unified way.
+
+After fileset is created, users can easily access, manage the files/directories through
+Fileset's identifier, without needing to know the physical path of the managed datasets. Also, with
+unified access control mechanism, filesets can also be managed via the same role based access
+control mechanism without needing to set access controls to different storages.
+
+To use fileset, we assume that:
```
Comment (qqqttt123) [inline docs/manage-fileset-metadata-using-gravitino.md:25]: Could we avoid using `we`?

Comment (jerryshao) [reply docs/manage-fileset-metadata-using-gravitino.md:25]: OK, let me double confirm all the places using  "we".

File: docs/index.md [hunk 12]
```diff
@@ -72,6 +74,10 @@ Gravitino currently supports the following catalogs:
 Gravitino also provides an Iceberg REST catalog service for the Iceberg table format. See the
 [Iceberg REST catalog service](./iceberg-rest-service.md) for details.
```
Comment (qqqttt123) [inline docs/index.md:75]: Should we include the message between relational catalog and fileset catalog?
```
Gravitino also provides an Iceberg REST catalog service for the Iceberg table format. See the
[Iceberg REST catalog service](./iceberg-rest-service.md) for details.
```

Comment (jerryshao) [reply docs/index.md:75]: Can you explain more?

Comment (qqqttt123) [reply docs/index.md:75]: I think it's more reasonable
```
Gravitino currently supports the following catalogs:
**Relational catalogs:**
* [**Iceberg catalog**](./lakehouse-iceberg-catalog.md)
* [**Hive catalog**](./apache-hive-catalog.md)
* [**MySQL catalog**](./jdbc-mysql-catalog.md)
* [**PostgreSQL catalog**](./jdbc-postgresql-catalog.md)

**Fileset catalogs:**

* [**Hadoop catalog**](./hadoop-catalog.md)

Gravitino also provides an Iceberg REST catalog service for the Iceberg table format. See the
[Iceberg REST catalog service](./iceberg-rest-service.md) for details.

```

**Review by shaofengshi — APPROVED** [review]
LGTM

File: docs/manage-fileset-metadata-using-gravitino.md [hunk 13]
```diff
@@ -0,0 +1,413 @@
...
+Map<String, String> properties = ImmutableMap.<String, String>builder()
+    .put("location", "file:/tmp/root")
+    // Property "location" is optional, if specified all the managed fileset without
+    // specifying storage location will be stored under this location.
+    .build();
+
+Catalog catalog = gravitinoClient.createCatalog(
+    NameIdentifier.of("metalake", "catalog"),
+    Type.FILESET,
+    "hadoop", // provider, We support only support "hadoop".
```
Comment (qqqttt123) [inline docs/manage-fileset-metadata-using-gravitino.md:75]: We.

File: docs/manage-fileset-metadata-using-gravitino.md [hunk 14]
```diff
@@ -0,0 +1,413 @@
...
+  propertiesMap,
+);
+```
+
+</TabItem>
+</Tabs>
+
+Currently, Gravitino supports two **types** of the fileset:
+
+ - `MANAGED`: The storage location of the fileset is managed by Gravitino, when specified as
```
Comment (qqqttt123) [inline docs/manage-fileset-metadata-using-gravitino.md:270]: I feel that it's not accurate for `managed by Gravitino` vs `managed by users`.

File: docs/manage-fileset-metadata-using-gravitino.md [hunk 15]
```diff
@@ -0,0 +1,413 @@
...
+This page introduces how to manage fileset metadata by Gravitino. Fileset is a concept brought
+out in Gravitino, which is a collection of files and directories. Users can leverage
+fileset to manage non-tabular data like training datasets, raw data.
+
+Typically, a fileset is mapping to a directory on a file system like HDFS, S3, ADLS, GCS, etc.
+With fileset managed by Gravitino, the non-tabular data can be managed as assets together with
+tabular data and others in Gravitino with a unified way.
+
+After fileset is created, users can easily access, manage the files/directories through
+Fileset's identifier, without needing to know the physical path of the managed datasets. Also, with
```
Comment (xloya) [inline docs/manage-fileset-metadata-using-gravitino.md:21]: Maybe `of the managed datasets` is not necessary.

File: docs/manage-fileset-metadata-using-gravitino.md [hunk 16]
```diff
@@ -0,0 +1,413 @@
...
+Please refer to [Alter a catalog](./manage-relational-metadata-using-gravitino.md#alter-a-catalog)
+from relational catalog for more details. For fileset catalog, the alter operation is the same.
+
+### Drop a catalog
+
+Please refer to [Drop a catalog](./manage-relational-metadata-using-gravitino.md#drop-a-catalog)
+from relational catalog for more details. For fileset catalog, the drop operation is the same.
+
+:::note
+Currently, Gravitino doesn't support dropping a catalog with schema and files under it. You have
```
Comment (xloya) [inline docs/manage-fileset-metadata-using-gravitino.md:106]: `files` -> `filesets`?

File: docs/manage-fileset-metadata-using-gravitino.md [hunk 17]
```diff
@@ -0,0 +1,413 @@
...
+to drop all the schemas and files under the catalog before dropping the catalog.
```
Comment (xloya) [inline docs/manage-fileset-metadata-using-gravitino.md:107]: Ditto.

**Review by qqqttt123 — APPROVED** [review]
LGTM.

File: docs/manage-fileset-metadata-using-gravitino.md (hunk 15 above)
Comment (jerryshao) [reply docs/manage-fileset-metadata-using-gravitino.md:21]: I feel like it is still necessary, it means that the dataset is managed by Gravitino, so users don't need to know the physical path. Fro unmanaged dataset, users still need to know the physical path before visiting the dataset.

Comment (xloya) [reply docs/manage-fileset-metadata-using-gravitino.md:21]: Got it.

**Pull request merged**